lib/app-common.ts

```diff
@@ -115,35 +115,6 @@ export interface GlobalTunnelNgConfig {
  * default exclusion patterns are added for all private IPv4 address ranges.
  */
 async function setupGlobalHttpProxy(settings: CliSettings) {
-	const semver = await import('semver');
-	if (semver.lt(process.version, '10.16.0')) {
-		setupGlobalTunnelNgProxy(settings);
-	} else {
-		// use global-agent instead of global-tunnel-ng
-		await setupGlobalAgentProxy(settings);
-	}
```

Comment:
I liked better having the separate `setupGlobalTunnelNgProxy` and `setupGlobalAgentProxy` functions, mainly because it makes it clearer / obvious that some proxy functionality only applies to each of them; for example, `BALENARC_NO_PROXY ` only applies to `setupGlobalAgentProxy`. Also, the suggested modification in this PR still runs a bit of unnecessary code if the version of Node is older than 10.16.0. What do you think of keeping the separate functions unmodified, and simply adding the following `if` clause at the top of the original `setupGlobalHttpProxy` function?

```
async function setupGlobalHttpProxy(settings: CliSettings) {
	if (
		!process.env.HTTP_PROXY &&
		!process.env.HTTPS_PROXY &&
		!process.env.http_proxy &&
		!process.env.https_proxy &&
		!settings.getCatch<string | GlobalTunnelNgConfig>('proxy')
	) {
		return;
	}
	... // the rest of the function is kept unmodified
```

This `if` clause matches the documented CLI proxy capabilities:
https://www.balena.io/docs/reference/balena-cli/#proxy-support